### PR TITLE
Ref #8918 - Enable creating container via katello

### PR DESCRIPTION
**Pull request by parthaa**
This commit provides the integration point to enable katello provision
docker container

### CODE REVIEW

File: app/views/containers/steps/image.html.erb [hunk 1]
```diff
@@ -3,7 +3,16 @@
 
 <%= render :layout => 'title', :locals => { :step => 2 } do %>
     <ul class="nav nav-tabs" data-tabs="tabs">
-      <li class="active"><a href="#hub" data-toggle="tab" id="hub_tab">
+      <% active_class = "active" -%>
+      <% if defined? Katello -%>
+        <% active_class = "" -%>
+        <li class="active"><a href="#katello" data-toggle="tab" id="katello_tab">
+          <span class="glyphicon glyphicon-cloud-download"></span>
```
Comment (eLobato) [inline app/views/containers/steps/image.html.erb]: May I suggest .glyphicon-tower? :smile:

**Comment by eLobato** [comment]
Green light here, @daviddavis might want to ack too. My only nitpick is the icon, users will probably appreciate some distinction between the tabs. Cool stuff!

File: app/views/containers/steps/image.html.erb (hunk 1 above)
Comment (daviddavis) [inline app/views/containers/steps/image.html.erb]: :+1:

**Comment by eLobato** [comment]
Merged as b747db40e14e9dba89a8df5a9f2f09df3165cc03, thanks @parthaa !

**Comment by eLobato** [comment]
oops, that message was meant for the docker pull PR, sorry.

**Comment by eLobato** [comment]
@parthaa this needs a rebase

**Comment by parthaa** [comment]
@eLobato should have been rebased. Please let me know if you find issues.

File: app/models/docker_container_wizard_state.rb [hunk 2]
```diff
@@ -15,6 +15,7 @@ def container_attributes
     { :repository_name     => image.repository_name,
       :tag                 => image.tag,
       :registry_id         => image.registry_id,
+      :katello             =>  image.katello?,
```
Comment (daviddavis) [inline app/models/docker_container_wizard_state.rb]: misaligned

File: app/views/containers/steps/image.html.erb [hunk 3]
```diff
@@ -2,67 +2,52 @@
...
+    katello_class = ""
+    hub_class = ""
+    registry_class = ""
+    if @docker_container_wizard_states_image.katello?
+      katello_class = " active"
+    elsif @docker_container_wizard_states_image.registry_id.nil?
+      hub_class = " active"
+    else
+      registry_class = " active"
+    end
```
Comment (daviddavis) [inline app/views/containers/steps/image.html.erb]: This code should be moved to a helper.

Comment (parthaa) [inline app/views/containers/steps/image.html.erb]: Are you sure ? Its just choosing whether a given class is active or not.

Comment (parthaa) [inline app/views/containers/steps/image.html.erb]: something for me to look at I guess..

**Comment by daviddavis** [comment]
Code looks good now. Two small items from testing:

1. When I enter in invalid info on the external registries tab and click submit, the docker hub tab becomes active.
2. [The Katello icon](https://www.dropbox.com/s/lvh1y98nmb0fzaz/Screenshot%202015-02-05%2013.26.04.png?dl=0) doesn't show up properly for me.

**Comment by parthaa** [comment]
> When I enter in invalid info on the external registries tab and click submit, the docker hub tab becomes active.

Can ya give me a couple of invalid options...

**Comment by daviddavis** [comment]
Enter in a repo but no tag.

**Comment by daviddavis** [comment]
@parthaa talked with you on IRC but I think we can open separate bugs for these items and just merge this as-is. Let me know.

**Comment by daviddavis** [comment]
ACK

**Comment by daviddavis** [comment]
Merged as 41572b90604e65a5cef552ea5943cb1ce1a0275d. :+1:

**Pull request closed without merging**